12_rpn_calculator/rpn_calculator.rb

```diff
@@ -51,19 +47,65 @@ def tokens(string)
 
   def evaluate(string)
     tokens(string).map do |value|
-      value.is_a?(Integer) ? array.push(value) : do_operation(value)
+      if value.is_a?(Integer)
+        array.push(value)
+      else
+        set_operation(value)
+        execute_operation
```

Comment:
Why do you do this in 2 steps ? Doing things this way on a shared object in a multi-threaded environment might end up with invalid results. Prefer as much as possible https://www.sitepoint.com/functional-programming-pure-functions/